code/modules/shareddream/dream_entry.dm

```diff
@@ -19,6 +19,10 @@ var/list/dream_entries = list()
 			to_chat(bg, "<span class='warning'>Whilst in shared dreaming, you find it difficult to hide your secrets.</span>")
 			if(willfully_sleeping)
 				to_chat(bg, "To wake up, use the \"Awaken\" verb in the IC tab.")
+			for(var/mob/living/carbon/human/H in living_mob_list)
+				if(H.can_commune())
+					to_chat(H, SPAN_CULT("You sense an increase in the activity of Srom..."))
+					sound_to(H, sound('sound/effects/psi/power_used.ogg'))
```

Comment:
The list of psionics is usually much smaller than the living mob list. What I would do here is instead:
- When a mob has psionics added to it, add this mob to a list on SSpsionics called `psionic_mobs` or something.
- Instead of iterating over the entire living mob list, iterate over `psionic_mobs` instead.

Reply:
had a look at SSpsi, and SSpsi.processing stores the psi datums for all of the mobs that *would* be in the created "psionic_mobs" list -- figure rather than creating a new list inside SSpsi that stores more or less the same data, it's simpler to just iterate through the existing list of datums and get their owner

this is also the method used in a few other areas, so i figured it's hopefully acceptable